### PR TITLE
Scan friends' QR code and more product info

### PR DESCRIPTION
**Pull request by ozan-guven**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/healthplay/BarcodeScanFragment.java [hunk 1]
```diff
@@ -170,17 +181,53 @@ private void scan(Uri uri) throws IOException {
...
+                                                usernameText.setText(username);
+                                                String pPicture = (String) user.get("image");
+                                                Glide.with(this).load(pPicture).into((ImageView) view.findViewById(R.id.profilePictureAddFriend));
+
+                                                view.findViewById(R.id.addFriendScanButton).setOnClickListener(onClick -> {
+                                                    // Add the user to the friend list
+                                                    database.addToFriendList(barcodeString);
+                                                    // Remove info view
+                                                    view.findViewById(R.id.shadowFrameScanFriend).setVisibility(View.GONE);
+                                                    Toast t = Toast.makeText(getContext(), username+" was added in your friends list !", Toast.LENGTH_SHORT);
```
Comment (emonnin-epfl) [inline app/src/main/java/ch/epfl/sdp/healthplay/BarcodeScanFragment.java:222]: no need to create a variable t if you use it right after
(Toast.makeText(getContext(), username+" was added in your friends list !", Toast.LENGTH_SHORT).show();)

File: app/src/main/java/ch/epfl/sdp/healthplay/model/Product.java [hunk 2]
```diff
@@ -211,6 +214,41 @@ public Nutriscore getNutriscore() {
...
+        }
+    }
+
+    public String getAllergens() {
+        try {
+            JSONArray array = product.getJSONArray("allergens_tags");
+            StringJoiner joiner = new StringJoiner(", ");
+            for (int i = 0; i < array.length(); i++) {
+                String name = array.getString(i);
+                joiner.add(name.substring(3));
```
Comment (emonnin-epfl) [inline app/src/main/java/ch/epfl/sdp/healthplay/model/Product.java:231]: Why 3?

**Review by emonnin-epfl — CHANGES_REQUESTED** [review]

**Review by Aco-Hub — APPROVED** [review]

File: app/src/main/java/ch/epfl/sdp/healthplay/model/Product.java (hunk 2 above)
Comment (ozan-guven) [reply app/src/main/java/ch/epfl/sdp/healthplay/model/Product.java:231]: This is to trim the language in front of the ingredient. I should make it clearer.

File: app/src/main/java/ch/epfl/sdp/healthplay/BarcodeScanFragment.java (hunk 1 above)
Comment (ozan-guven) [reply app/src/main/java/ch/epfl/sdp/healthplay/BarcodeScanFragment.java:222]: You are right, will change this !

File: app/src/main/java/ch/epfl/sdp/healthplay/model/Product.java (hunk 2 above)
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/model/Product.java]: OK, great!

**Review by emonnin-epfl — APPROVED** [review]

**Pull request merged**